ksql-engine/src/main/java/io/confluent/ksql/util/QueryIdGenerator.java

```diff
@@ -15,25 +15,12 @@
 
 package io.confluent.ksql.util;
 
-import java.util.concurrent.atomic.AtomicLong;
-
-public class QueryIdGenerator {
-
-  private final AtomicLong queryIdCounter;
-
-  public QueryIdGenerator() {
-    this(0L);
-  }
-
-  private QueryIdGenerator(final long initialValue) {
-    this.queryIdCounter = new AtomicLong(initialValue);
-  }
+/**
+ * Generator used to provide query ids
+ */
+public interface QueryIdGenerator {
 
-  public String getNextId() {
-    return String.valueOf(queryIdCounter.getAndIncrement());
-  }
+  String getNextId();
```

Comment:
Consider just making this:

```suggestion
  long getNext();
```

Which will fit nicely with my proposed:

```java
    long peekNext();
```